src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/perform_interaction_request.cc

```diff
@@ -1127,43 +1117,58 @@ mobile_apis::Result::eType
 PerformInteractionRequest::PrepareResultCodeForResponse(
     const app_mngr::commands::ResponseInfo& ui_response,
     const app_mngr::commands::ResponseInfo& vr_response) {
-  LOG4CXX_DEBUG(logger_,
-                "InteractionMode = " << static_cast<int32_t>(interaction_mode_)
-                                     << " | FirstAnsweredInterface = "
-                                     << static_cast<int32_t>(first_responder_));
-
-  if (mobile_apis::InteractionMode::VR_ONLY == interaction_mode_) {
-    if (FirstAnsweredInterface::VR == first_responder_) {
-      return MessageHelper::HMIToMobileResult(vr_result_code_);
+  LOG4CXX_DEBUG(
+      logger_, "InteractionMode = " << static_cast<int32_t>(interaction_mode_));
+
+  auto result = mobile_api::Result::INVALID_ENUM;
+  switch (interaction_mode_) {
```

Comment:
I think a switch statement makes this more difficult to read, if blocks with a return in them are easier to follow